apps/communications/contacts/js/views/details.js

```diff
@@ -320,6 +324,11 @@ contacts.Details = (function() {
   };
 
   var toggleFavorite = function toggleFavorite() {
+    // Because we are caching the first chunk of contacts and the favorites are
+    // the contacts that are first shown in the screen, we need to evict the
+    // cache if any change is done in the favorites list.
+    Cache.evict();
```

Comment:
I think we don't need to evict here, since this function will perform an `save` and that will trigger a `oncontactchange`